### PR TITLE
WIP: Configurable load balancer attributes (access logging and read timeout)

### LINKED ISSUE #90: Enable access logging for ALBs created with alb-ingress-controller
Hello,

We'd like to enable access logging for ALBs that alb-ingress-controller create. We are currently doing this manually for long-running services, but it would be nice if we could configure that in alb-ingress-controller.

ALBs support logging to S3, see https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#enable-access-logging

### PR DESCRIPTION
**Pull request by antoniobeyah**
Fixes #90 (access logging) and #79 (read timeout)

### CODE REVIEW

File: pkg/annotations/annotations.go [hunk 1]
```diff
@@ -106,6 +109,33 @@ func ParseAnnotations(annotations map[string]string, clusterName string) (*Annot
...
+	var attrs []*elbv2.LoadBalancerAttribute
+	var badAttrs []string
+	rawAttrs := util.NewAWSStringSlice(annotations[attributesKey])
+
+	for _, rawAttr := range rawAttrs {
+		parts := strings.Split(*rawAttr, "=")
+		switch {
+		case *rawAttr == "":
+			continue
+		case len(parts) < 2:
```
Comment (joshrosso) [inline pkg/annotations/annotations.go]: You'll always expect `parts` to have length `2` here right?

Possibly best to set this check such that when it's anything other than 2, we return the error. Preventing cases like `attr=value=value2`

**Comment by antoniobeyah** [comment]
@joshrosso how should the controller deal with configuration items there were set outside of the controller?  Looking at the way the reconcile works if someone setup access logging outside the scope of the controller and did not add in the matching annotations access logging would be disabled.  Is this something we should try to handle in code or just add it as a warning in the changelog?

**Comment by joshrosso** [comment]
@antoniobeyah Our approach thus far has been that all ALBs the controller manages always uses the desired state (derived from ingress manifests) as what we'll eventually get to. Meaning, any configuration options made outside of the ALB (ie AWS) should be overwritten by the controller in the next reconcile.

Let me know if this answers your Q.

**Comment by masterzen** [comment]
Hi,

Because we needed to setup the ALB access log, I finished the work started by @antoniobeyah in this PR. Since it is not possible to contribute directly to this PR, I've open a new one: #279 .

I've tested the feature on a live k8s cluster.

Please let me know if there's anything I need to do.
Thanks
Brice

**Pull request closed without merging**